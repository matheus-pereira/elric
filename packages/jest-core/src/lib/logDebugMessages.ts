/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import type {Config} from '@elric/types';
const VERSION = require('../../package.json').version;

// if the output here changes, update `getConfig` in e2e/runelric.ts
export default function logDebugMessages(
  globalConfig: Config.GlobalConfig,
  configs: Array<Config.ProjectConfig> | Config.ProjectConfig,
  outputStream: NodeJS.WriteStream,
): void {
  const output = {
    configs,
    globalConfig,
    version: VERSION,
  };
  outputStream.write(JSON.stringify(output, null, '  ') + '\n');
}
