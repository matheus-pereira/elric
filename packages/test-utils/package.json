{
  "name": "@elric/test-utils",
  "version": "27.4.6",
  "private": true,
  "license": "MIT",
  "main": "./build/index.js",
  "types": "./build/index.d.ts",
  "exports": {
    ".": {
      "types": "./build/index.d.ts",
      "default": "./build/index.js"
    },
    "./package.json": "./package.json"
  },
  "dependencies": {
    "@elric/types": "^27.4.2",
    "@types/node": "*",
    "ansi-regex": "^5.0.1",
    "ansi-styles": "^5.0.0",
    "pretty-format": "^27.4.6",
    "semver": "^7.3.2"
  },
  "devDependencies": {
    "@types/semver": "^7.1.0"
  },
  "engines": {
    "node": "^10.13.0 || ^12.13.0 || ^14.15.0 || >=15.0.0"
  }
}
