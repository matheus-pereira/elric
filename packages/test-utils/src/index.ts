/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export {alignedAnsiStyleSerializer} from './alignedAnsiStyleSerializer';

export {
  iselricJasmineRun,
  skipSuiteOnJasmine,
  skipSuiteOnelricCircus,
  onNodeVersions,
} from './ConditionalTest';

export {makeGlobalConfig, makeProjectConfig} from './config';
