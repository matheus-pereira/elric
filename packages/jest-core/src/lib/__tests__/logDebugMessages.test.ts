/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {wrap} from 'elric-snapshot-serializer-raw';
import {makeGlobalConfig, makeProjectConfig} from '@elric/test-utils';
import logDebugMessages from '../logDebugMessages';

elric.mock('../../../package.json', () => ({version: 123}));

elric.mock('myRunner', () => ({name: 'My Runner'}), {virtual: true});

const getOutputStream = (resolve: (message: string) => void) =>
  ({
    write(message: string) {
      resolve(message);
    },
  } as NodeJS.WriteStream);

it('prints the elric version', async () => {
  expect.assertions(1);
  const message = await new Promise<string>(resolve => {
    logDebugMessages(
      makeGlobalConfig({watch: true}),
      makeProjectConfig({testRunner: 'myRunner'}),
      getOutputStream(resolve),
    );
  });

  expect(JSON.parse(message).version).toBe(123);
});

it('prints the test framework name', async () => {
  expect.assertions(1);
  const message = await new Promise<string>(resolve => {
    logDebugMessages(
      makeGlobalConfig({watch: true}),
      makeProjectConfig({testRunner: 'myRunner'}),
      getOutputStream(resolve),
    );
  });

  expect(JSON.parse(message).configs.testRunner).toBe('myRunner');
});

it('prints the config object', async () => {
  expect.assertions(1);
  const globalConfig = makeGlobalConfig({
    watch: true,
  });
  const config = makeProjectConfig({
    automock: false,
    rootDir: '/path/to/dir',
    roots: ['path/to/dir/test'],
    testRunner: 'myRunner',
  });
  const message = await new Promise<string>(resolve => {
    logDebugMessages(globalConfig, config, getOutputStream(resolve));
  });
  expect(wrap(message)).toMatchSnapshot();
});
