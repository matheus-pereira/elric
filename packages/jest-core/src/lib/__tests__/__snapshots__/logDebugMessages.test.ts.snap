// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`prints the config object 1`] = `
{
  "configs": {
    "automock": false,
    "cache": false,
    "cacheDirectory": "/test_cache_dir/",
    "clearMocks": false,
    "coveragePathIgnorePatterns": [],
    "cwd": "/test_root_dir/",
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "extensionsToTreatAsEsm": [],
    "extraGlobals": [],
    "forceCoverageMatch": [],
    "globals": {},
    "haste": {},
    "injectGlobals": true,
    "moduleDirectories": [],
    "moduleFileExtensions": [
      "js"
    ],
    "moduleLoader": "/test_module_loader_path",
    "moduleNameMapper": [],
    "modulePathIgnorePatterns": [],
    "modulePaths": [],
    "name": "test_name",
    "prettierPath": "prettier",
    "resetMocks": false,
    "resetModules": false,
    "restoreMocks": false,
    "rootDir": "/path/to/dir",
    "roots": [
      "path/to/dir/test"
    ],
    "runner": "elric-runner",
    "setupFiles": [],
    "setupFilesAfterEnv": [],
    "skipFilter": false,
    "skipNodeResolution": false,
    "slowTestThreshold": 5,
    "snapshotFormat": {},
    "snapshotSerializers": [],
    "testEnvironment": "node",
    "testEnvironmentOptions": {},
    "testLocationInResults": false,
    "testMatch": [],
    "testPathIgnorePatterns": [],
    "testRegex": [
      "\\\\.test\\\\.js$"
    ],
    "testRunner": "myRunner",
    "testURL": "http://localhost",
    "timers": "real",
    "transform": [],
    "transformIgnorePatterns": [],
    "watchPathIgnorePatterns": []
  },
  "globalConfig": {
    "bail": 0,
    "changedFilesWithAncestor": false,
    "changedSince": "",
    "collectCoverage": false,
    "collectCoverageFrom": [],
    "coverageDirectory": "coverage",
    "coverageProvider": "babel",
    "coverageReporters": [],
    "coverageThreshold": {
      "global": {}
    },
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "expand": false,
    "findRelatedTests": false,
    "forceExit": false,
    "json": false,
    "lastCommit": false,
    "listTests": false,
    "logHeapUsage": false,
    "maxConcurrency": 5,
    "maxWorkers": 2,
    "noStackTrace": false,
    "nonFlagArgs": [],
    "notify": false,
    "notifyMode": "failure-change",
    "onlyChanged": false,
    "onlyFailures": false,
    "passWithNoTests": false,
    "projects": [],
    "reporters": [],
    "rootDir": "/test_root_dir/",
    "runTestsByPath": false,
    "silent": false,
    "skipFilter": false,
    "snapshotFormat": {},
    "testFailureExitCode": 1,
    "testNamePattern": "",
    "testPathPattern": "",
    "testSequencer": "@elric/test-sequencer",
    "testTimeout": 5000,
    "updateSnapshot": "none",
    "useStderr": false,
    "verbose": false,
    "watch": true,
    "watchAll": false,
    "watchPlugins": [],
    "watchman": false
  },
  "version": 123
}

`;
